src/sentry/middleware/auth.py

```diff
@@ -105,11 +104,11 @@ def process_request(self, request: HttpRequest) -> None:
                     request.user, request.auth = result
                 else:
                     # default to anonymous user and use IP ratelimit
-                    request.user = SimpleLazyObject(lambda: get_user(request))
```

Comment:
Looking at the [source for LazyObject](https://github.com/django/django/blob/stable/3.2.x/django/utils/functional.py#L251) I don't see much benefit for us to have the request user be lazy wrapped as we *always* use the request user to build `request.access` which would negate the benefits of lazy instantiation.